runtime/src/accounts_db.rs

```diff
@@ -9916,87 +9967,229 @@ pub mod tests {
+            common_slot_id,
+            store3_id,
+            store_file_size,
         ));
-        entry2.alive_bytes.store(3000, Ordering::Relaxed);
+
+        // The store3's alive ratio is 1.0 as its page-aligned alive size is 2 pages
+        let store3_alive_bytes = 1024 * 7;
+        entry3
+            .alive_bytes
+            .store(store3_alive_bytes, Ordering::Relaxed);
 
         candidates
-            .entry(dummy_slot2)
+            .entry(common_slot_id)
             .or_default()
-            .insert(entry2.append_vec_id(), entry2.clone());
+            .insert(entry3.append_vec_id(), entry3.clone());
 
-        let output_candidates =
-            accounts.select_candidates_by_total_usage(&candidates, DEFAULT_ACCOUNTS_SHRINK_RATIO);
-        assert_eq!(2, output_candidates.len());
-        assert_eq!(1, output_candidates[&dummy_slot].len());
-        assert_eq!(1, output_candidates[&dummy_slot2].len());
+        // Set the target ratio to default (0.8), both store1 and store2 must be selected and store3 is ignored.
+        let target_alive_ratio = DEFAULT_ACCOUNTS_SHRINK_RATIO;
+        let (selected_candidates, next_candidates) =
+            accounts.select_candidates_by_total_usage(&candidates, target_alive_ratio);
+        assert_eq!(1, selected_candidates.len());
+        assert_eq!(2, selected_candidates[&common_slot_id].len());
+        assert!(selected_candidates[&common_slot_id].contains(&store1.append_vec_id()));
+        assert!(selected_candidates[&common_slot_id].contains(&store2.append_vec_id()));
+        assert_eq!(0, next_candidates.len());
+    }
+
+    #[test]
+    fn test_select_candidates_by_total_usage_all_clean() {
+        // 2 candidates, they must be selected to achieve the target alive ratio
+        solana_logger::setup();
+        let accounts = AccountsDb::new_single_for_tests();
+        let mut candidates: ShrinkCandidates = HashMap::new();
+
+        let slot1 = 12;
+        let common_store_path = Path::new("");
+
+        let store_file_size = 4 * PAGE_SIZE;
+        let store1_id = 22;
+        let store1 = Arc::new(AccountStorageEntry::new(
+            common_store_path,
+            slot1,
+            store1_id,
+            store_file_size,
+        ));
+
+        // store1 has 1 page-aligned alive bytes, its alive ratio is 1/4: 0.25
+        let store1_alive_bytes = 3 * 1024;
+        store1
+            .alive_bytes
+            .store(store1_alive_bytes, Ordering::Relaxed);
+
+        candidates
+            .entry(slot1)
+            .or_default()
+            .insert(store1.append_vec_id(), store1.clone());
+
+        let store2_id = 44;
+        let slot2 = 44;
+        let store2 = Arc::new(AccountStorageEntry::new(
+            common_store_path,
+            slot2,
+            store2_id,
+            store_file_size,
+        ));
+
+        // store2 has 2 page-aligned bytes, its alive ratio is 2/4: 0.5
+        let store2_alive_bytes = 5 * 1024;
```

Comment:
Can we just do `can we just use store_file_size / 2?

Reply:
I intentionally keep them not the rounded to the page size so that we are examining the logic is doing page aligned calculations

Reply:
Good idea